### ISSUE TITLE
NotSupportedException when run a simple script on new machine

### ISSUE THREAD

**Issue by constructor-igor** [cake-build/cake#1316]
cake script (file build.cake):

```
#tool "nuget:?package=NUnit.Runners&version=2.6.4"

Task("Test")
    .Does(()=>
{
    Information("Test");
});

RunTarget("Test");
```

build.cmd

```
@echo off
powershell ./build.ps1 %CAKE_ARGS% %* -Verbosity Diagnostic
pause
```

![image](https://cloud.githubusercontent.com/assets/1849690/19676112/03ea9798-9a9b-11e6-8d34-507face520c6.png)

**Comment by gep13** [comment]
@constructor-igor what version of Cake are you using?

**Comment by constructor-igor** [comment]
0.16.2.0

![image](https://cloud.githubusercontent.com/assets/1849690/19676401/e66d2fbc-9a9c-11e6-8e06-c9807bdd476b.png)

**Comment by gep13** [comment]
And what command are you executing when you get that output?  Are you passing any arguments to your `build.cmd`

**Comment by constructor-igor** [comment]
I call ```build.cmd``` without parameters.

**Comment by devlead** [comment]
What happens if you'r in a folder not named `@` i.e. `d:\temp`

**Comment by constructor-igor** [comment]
in the case (folder ```d:\temp```) cake works without problems.

**Comment by devlead** [comment]
Ok then probably similar to #1245

**Comment by patriksvensson** [comment]
Should we close this in favor of #1245?

**Comment by gep13** [comment]
@patriksvensson #1245 has been closed as fixed.  There was some discussion about creating another issue to make the globbing more resilient, but I don't know if that got created or not.

**Comment by stffabi** [comment]
@gep13 i've just fixed one of such case with #1321. I could pick that task and try to make it more resilient if you wish.

**Comment by gep13** [comment]
@stffabi sure, I would have no objection to this.

Perhaps best to discuss here your intended implementation, before actually doing anything.

**Comment by stffabi** [comment]
@gep13 I'll write down a little proposal for the weekend.

**Comment by gep13** [comment]
@stffabi sounds great!

**Comment by stffabi** [comment]
I think the best would be to make the GlobTokenizer resilient against invalid file/path characters. The tokenizer shouldn't be responsible for any file/path character checking, that's the job of the filesystem. It should be independent of the filesystem and not have any filesystem dependent code in it. E.g. currently there's a `WindowsRoot` token, in order to support this, the tokenizer mainly prevents the use of  `:` in a path on an unix system. The Globber definition has some special characters, `*`, `?`, `/`, `\`. Apart from these the tokenizer should allow all characters to be used in an identifier and later on the specific filesystem should decide if it's valid or not.

@gep13 what do you think?

**Comment by JeremyCade** [comment]
There is a #1343  that fixes this issue and it's duplicate #1342.  

May I suggest that it is merged in the interim to avoid issues with Jenkins CI Pipeline builds, given that that Jenkins Workspace directories in the Pipeline Multibranch model almost always have @ in the directory name. It is a significant pain point.

**Comment by devlead** [comment]
Fixed by #1343

**Comment by stffabi** [comment]
@devlead didn't we want to make the GlobTokenizer more resilient? I'm still waiting on a reply from @gep13 on my proposal.

Should I create a new issue for making the GlobTokenizer more resilient?

**Comment by gep13** [comment]
@stffabi yes, I think another issue would make sense.  This issue was a specific failure of the Globber, which is now addressed, but there is still another, larger, underlying issue.  Your proposal sounds fine to me.  Would you be in a position to do a PR to provide an implementation, so that we can discuss further?

**Comment by stffabi** [comment]
@gep13 that's fine. I've created theissue #1354, furthermore I've also started with a PR.
